### PR TITLE
Document HSM usage for members identity and encryption keys

### LINKED ISSUE #1844: Proposal signature with key held in HSM
As members' identity keys may be held in a HSM, proposals should be able to be signed with such keys. CCF currently only supports `ecdsa-sha256` for the HTTP signature scheme. However, Azure HSMs don't seem to support this signature algorithm (see https://docs.microsoft.com/en-us/rest/api/keyvault/sign/sign#jsonwebkeysignaturealgorithm) but only SHA-384 on the 384r1 EC curve.

With regards to the type of signature algorithm we should support, https://tools.ietf.org/html/draft-cavage-http-signatures-12#section-2.1.3 states:

> Implementers SHOULD derive the digital signature algorithm used by an implementation from the key metadata identified by the `keyId` rather than from this field.  If `algorithm` is provided and differs from the key metadata identified by the `keyId`, for example `rsa-sha256` but an EdDSA key is identified via `keyId`, then an implementation MUST produce an error.

> Algorithm Name: ecdsa-sha256
   Status: deprecated, specifying signature algorithm enables attack vector.

> Algorithm Name: hs2019
   Status: active

A solution would then be to sign the http request using P-384 and SHA-384, use the `hs2019` algorithm in the HTTP header and derive the hashing algorithm from the member's identity key retrieved from `keyId`.

Note that the Python package used to sign HTTP requests ([`requests-http-signature`](https://github.com/pyauth/requests-http-signature)) doesn't support `hs2019` and only supports `sha256` as the signing algorithm so we would have to contribute these changes upstream or remove the dependency on this package.

### LINKED ISSUE #1845: Document use of Azure HSM to store member identity and encryption keys
Follow-up from https://github.com/microsoft/CCF/issues/1720#issuecomment-718878854, we should test and document how members keys can be held in AKV, registered in CCF and used for signing proposals and decrypting recovery shares.

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #1844
Resolves #1845 

This PR adds documentation for how members can store their identity keys/certificates and encryption keys inside Azure Key Vault and how to use them with CCF. 

Most changes are documentation, except a minor change to `scurl.sh` which now prints the digest to be signed as well as the request headers if `--print-digest-to-sign. This is useful when constructing signed requests with a HSM (see new documentation page for an example)..

### CODE REVIEW

**Review by achamayou — APPROVED** [review]
Looks good!

**Comment by achamayou** [comment]
One of the json files needs reformatting.

File: doc/members/adding_member.rst [hunk 1]
```diff
@@ -3,11 +3,13 @@ Adding New Members
 
 It is possible for existing members to add new members to the consortium after a CCF network has been started.
 
-.. note:: The maximum number of active consortium members at any given time is 255.
+.. note:: The maximum number of active recovery members (i.e. those with a recovery share) at any given time is 255.
```
Comment (ashamis) [inline doc/members/adding_member.rst:6]: you probably want to say something like "The maximum number of allowed active ...."

File: doc/members/hsm_keys.rst [hunk 2]
```diff
@@ -0,0 +1,126 @@
+Using Member Keys Stored in HSM
+===============================
+
+This page explains how members' identity certificates and encryption keys stored in an `HSM <https://en.wikipedia.org/wiki/Hardware_security_module>`_ can be used with CCF. The following guide describes the usage of `Azure Key Vault <https://azure.microsoft.com/en-gb/services/key-vault>`_
+
+.. note::
+
+    It is assumed that the CCF member has already access to an existing Azure Key Vault. See `here <https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-portal#create-a-vault>`_ for more details on how to create one. Using the `Azure CLI <https://docs.microsoft.com/en-us/cli/azure/install-azure-cli>`_, it is possible to check the list of available Key Vault instances:
```
Comment (ashamis) [inline doc/members/hsm_keys.rst:8]: "that CCF members already have"

**Review by ashamis — APPROVED** [review]

**Pull request merged**